### ISSUE TITLE
Nonexisting member not reported for class inheriting pyqt's QObject

### ISSUE THREAD

**Issue by PowerSnail** [microsoft/pyright#2338]
**Describe the bug**

If class inherits `QObject` from `PyQt.QtCore`, pyright does not report any error when accessing non-existing member.

**To Reproduce**

Run pyright on the code below.

**Expected behavior**

An error should be reported, as any other classes.

**Screenshots or Code**

```python
# inherit_qobject.py
from PyQt6.QtCore import QObject

class A(QObject):
    def __init__(self):
        super().__init__()

a = A()
a.gibberish
```

```bash
> pyright inherit_qobject.py 

...
0 errors, 0 warnings, 0 infos
...
```

```python
# normal_class.py
class A:
    def __init__(self):
        ...

a = A()
a.gibberish
```

```bash
> pyright normal_class.py

...
/path/to/normal_class.py
  /path/to/normal_class.py:8:3 - error: Cannot access member "gibberish" for type "A"
    Member "gibberish" is unknown (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
...
```

**VS Code extension or command-line**

Command line. 

```bash
> pyright --version
pyright 1.1.170
```

**Additional context**

pyqt version: 6.1.1
python version: 3.9.6

Here's the relevant parts of the stubs shipped with PyQt:

```python
# QtCore.pyi

class QObject(PyQt6.sip.wrapper):

    staticMetaObject = ... # type: 'QMetaObject'

    def __init__(self, parent: typing.Optional['QObject'] = ...) -> None: ...

    @typing.overload
    @staticmethod
    def disconnect(a0: 'QMetaObject.Connection') -> bool: ...
    @typing.overload
    def disconnect(self) -> None: ...
    def isSignalConnected(self, signal: 'QMetaMethod') -> bool: ...
    def senderSignalIndex(self) -> int: ...
    def disconnectNotify(self, signal: 'QMetaMethod') -> None: ...
    def connectNotify(self, signal: 'QMetaMethod') -> None: ...
    def customEvent(self, a0: 'QEvent') -> None: ...
    def childEvent(self, a0: 'QChildEvent') -> None: ...
    def timerEvent(self, a0: 'QTimerEvent') -> None: ...
    def receivers(self, signal: PYQT_SIGNAL) -> int: ...
    def sender(self) -> 'QObject': ...
    def deleteLater(self) -> None: ...
    def inherits(self, classname: str) -> bool: ...
    def parent(self) -> 'QObject': ...
    def objectNameChanged(self, objectName: str) -> None: ...
    def destroyed(self, object: typing.Optional['QObject'] = ...) -> None: ...
    def property(self, name: str) -> typing.Any: ...
    def setProperty(self, name: str, value: typing.Any) -> bool: ...
    def dynamicPropertyNames(self) -> typing.List['QByteArray']: ...
    def dumpObjectTree(self) -> None: ...
    def dumpObjectInfo(self) -> None: ...
    def removeEventFilter(self, a0: 'QObject') -> None: ...
    def installEventFilter(self, a0: 'QObject') -> None: ...
    def setParent(self, a0: 'QObject') -> None: ...
    def children(self) -> typing.List['QObject']: ...
    def killTimer(self, id: int) -> None: ...
    def startTimer(self, interval: int, timerType: Qt.TimerType = ...) -> int: ...
    def moveToThread(self, thread: 'QThread') -> None: ...
    def thread(self) -> 'QThread': ...
    def blockSignals(self, b: bool) -> bool: ...
    def signalsBlocked(self) -> bool: ...
    def isWindowType(self) -> bool: ...
    def isWidgetType(self) -> bool: ...
    def setObjectName(self, name: str) -> None: ...
    def objectName(self) -> str: ...
    @typing.overload
    def findChildren(self, type: type, name: str = ..., options: Qt.FindChildOption = ...) -> typing.List['QObject']: ...
    @typing.overload
    def findChildren(self, types: typing.Tuple, name: str = ..., options: Qt.FindChildOption = ...) -> typing.List['QObject']: ...
    @typing.overload
    def findChildren(self, type: type, re: 'QRegularExpression', options: Qt.FindChildOption = ...) -> typing.List['QObject']: ...
    @typing.overload
    def findChildren(self, types: typing.Tuple, re: 'QRegularExpression', options: Qt.FindChildOption = ...) -> typing.List['QObject']: ...
    @typing.overload
    def findChild(self, type: type, name: str = ..., options: Qt.FindChildOption = ...) -> 'QObject': ...
    @typing.overload
    def findChild(self, types: typing.Tuple, name: str = ..., options: Qt.FindChildOption = ...) -> 'QObject': ...
    @staticmethod
    def tr(sourceText: str, disambiguation: typing.Optional[str] = ..., n: int = ...) -> str: ...
    def eventFilter(self, a0: 'QObject', a1: 'QEvent') -> bool: ...
    def event(self, a0: 'QEvent') -> bool: ...
    def __getattr__(self, name: str) -> typing.Any: ...
    def pyqtConfigure(self, a0: typing.Any) -> None: ...
    def metaObject(self) -> 'QMetaObject': ...
```

```python
# sip.pyi
class wrappertype: ...
class simplewrapper: ...
class wrapper(simplewrapper): ...
```

**Comment by erictraut** [comment]
This is because the `QObject` (in type stub `PyQt6/QtCore.pyi`) defines a `__getattr__` method:

```python
    def __getattr__(self, name: str) -> typing.Any: ...
```

If you think that this is an error, please report it to the maintainers of the PyQt6 library.
